### PR TITLE
qemu-dm-wrapper: Call a wrapper script for upstream QEMU if requested

### PR DESCRIPTION
**Pull request by dvrabel**

### CODE REVIEW

File: scripts/qemu-dm-wrapper [hunk 1]
```diff
@@ -13,94 +13,125 @@
...
-	proc = subprocess.Popen([ "/usr/bin/env", "PATH=%s" % path ] + args,stdin=None,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True)
-	rc = proc.wait()
-	stdout = proc.stdout
-	stderr = proc.stderr
-	return (rc, stdout, stderr)
-
-def xenstore_write(path, value):
-	(rc, stdout, stderr) = doexec([xenstore_write_cmd, path, value])
+    """Execute a subprocess, then return its return code, stdout and stderr"""
+    proc = subprocess.Popen(["/usr/bin/env", "PATH=%s" % path] + args,
```
Comment (lindig) [inline scripts/qemu-dm-wrapper]: The environment (including `PATH`) can be passed as a dictionary to `subprocess.Popen` such that using `env` is not necessary. https://docs.python.org/2/library/subprocess.html#popen-constructor

**Comment by jonludlam** [comment]
Just to confirm, we now have 2 qemu wrapper scripts? Ugh.

**Comment by dvrabel** [comment]
New script has a specific purpose (transforming the command line args to be upstream QEMU compatible). It needs to live with QEMU so it can be more easily kept in sync by ring0 team members.  It will be removed as part of removing support for qemu-trad.

**Comment by jonludlam** [comment]
OK, as long as we've got a plan to remove it, I'm happy :-)

**Pull request merged**